arch/arm/core/cortex_m/prep_c.c

```diff
@@ -24,6 +24,45 @@
 #include <arch/arm/cortex_m/cmsis.h>
 #include <string.h>
 
+extern uint32_t _interrupt_stack;
+
+static inline void switch_sp_to_psp(void)
+{
+	__set_CONTROL(__get_CONTROL() | CONTROL_SPSEL_Msk);
+	/* Synchronous barrier to make psp new SP effective immediately */
+	__ISB();
+}
+
+static inline void set_and_switch_to_psp(void)
+{
+	uint32_t process_sp = (uint32_t)&_interrupt_stack;
+	process_sp += CONFIG_ISR_STACK_SIZE;
+	__set_PSP(process_sp);
+	switch_sp_to_psp();
+}
+
+void lock_interrupts(void)
+{
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
+	__disable_irq();
+#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+	__set_BASEPRI(_EXC_IRQ_DEFAULT_PRIO);
+#else
+#error Unknown ARM architecture
+#endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
+}
+
+#ifdef CONFIG_INIT_STACKS
+static inline void init_stacks(void)
```

Comment:
I would rather define `init_stacks()` only if `CONFIG_INIT_STACKS `is defined; no need of empty function definition for a function that is only used in this file.
This means, of course that init_stacks shall only be invoked `#ifdef CONFIG_INIT_STACKS`.

Reply:
Sure, I will update it in next pull request.

Reply:
Looks like yo added the #ifdef, no?